src/test/java/io/inscopemetrics/kairosdb/aggregators/AbstractHistogramTest.java

```diff
@@ -41,56 +41,72 @@
-            final Collection<Object> parametersPerFactory) {
-        final List<Object[]> parameterSets = new ArrayList<>();
-        for (final CreateHistogramFromValues factory : getHistogramFactoriesFromValues()) {
-            final Object[] parameters = new Object[parametersPerFactory.size() + 1];
-
-            // The first parameter is the factory
-            parameters[0] = factory;
-
-            // The next parameters are the ones supplied
-            int i = 1;
-            for (final Object parameter : parametersPerFactory) {
-                parameters[i] = parameter;
-                ++i;
-            }
+    public static Collection<Object[]> createParameters() {
+        return createParameterization(ALL_PARAMETERS_BY_TYPE);
+    }
 
-            // Add the parameter set
-            parameterSets.add(parameters);
-        }
-        return parameterSets;
+    /**
+     * Create a parameterization per histogram type providing only from values
+     * factories.
+     *
+     * @return The resulting parameterization by factory.
+     */
+    public static Collection<Object[]> createParametersFromValues() {
+        return createParameterization(FROM_VALUES_PARAMETERS_BY_TYPE);
     }
 
     /**
-     * Create a parameterization per histogram from counts factory of the
-     * supplied parameters.
+     * Create a parameterization per histogram type providing only from counts
+     * factories.
      *
-     * @param parametersPerFactory The parameters to include with each factory.
      * @return The resulting parameterization by factory.
      */
-    public static Collection<Object[]> createParametersForHistogramFromCounts(
-            final Collection<Object> parametersPerFactory) {
-        final List<Object[]> parameterSets = new ArrayList<>();
-        for (final CreateHistogramFromCounts factory : getHistogramFactoriesFromCounts()) {
-            final Object[] parameters = new Object[parametersPerFactory.size() + 1];
+    public static Collection<Object[]> createParametersFromCounts() {
+        return createParameterization(FROM_COUNTS_PARAMETERS_BY_TYPE);
+    }
 
-            // The first parameter is the factory
-            parameters[0] = factory;
+    private static Collection<Object[]> createParameterization(
+            final List<List<Object>> parameterListA
+    ) {
+        final List<Object[]> parameterSets = new ArrayList<>();
+        for (final List<Object> factoryList : parameterListA) {
+            final Object[] parameters = new Object[factoryList.size()];
 
-            // The next parameters are the ones supplied
-            int i = 1;
-            for (final Object parameter : parametersPerFactory) {
-                parameters[i] = parameter;
-                ++i;
+            // The first n parameters are the factories
+            for (int i = 0; i < factoryList.size(); ++i) {
+                parameters[i] = factoryList.get(i);
             }
```

Comment:
So this is what I didn't mention (have added some TODO's since then). The flaw with this parameterization is that if you have a parameterized test it's not so easy to use. The next step here is to have each of the `create....()` variants accept a called specified parameterization and then to have the `createParameterization()` method multiply the two together. I started adding this early when I thought I needed to parameterize one of the tests in more ways than just the factory but it proved false. Nevertheless, I it's probably going to come up in this library.

Reply:
Is there some use-case that you already see this coming up for?

Spencer and I [chatted about TODO comments on one of his PRs](https://github.com/ArpNetworking/metrics-portal/pull/378#discussion_r424775806) and I'm of the opinion that if the timeline for resolving a TODO is nebulous it should probably be removed and replaced by filing an issue.

At the very least, could you file an issue for this?

Reply:
Sure I can. However, it's not something that should/needs to be fixed of its own accord. It's something that someone looking to parameterize one of these tests in additional ways will run to the comment and structure of the code recommends a path forward.

Reply:
https://github.com/InscopeMetrics/kairosdb-extensions/issues/32

Reply:
Thanks!